tests/test_install.py

```diff
@@ -0,0 +1,308 @@
+    nine_nodes_config_dict['ca_cert_path'] = ca_path
+    for node_name, node_dict in nine_nodes_config_dict['existing_vms'].items():
+        for val in 'key', 'cert':
+            val_path = tmp_certs_dir / '{0}_{1}.pem'.format(node_name, val)
+            val_path.write_text(u'{0}_{1}'.format(node_name, val))
+            node_dict['{0}_path'.format(val)] = str(val_path)
+
+    with mock.patch('cfy_cluster_manager.main.CERTS_DIR',
+                    str(cluster_manager_certs_dir)):
+        _generate_general_cluster_dict(nine_nodes_config_dict)
+
+    with mock.patch('cfy_cluster_manager.main.CA_PATH',
+                    str(cluster_manager_certs_dir / 'ca.pem')):
+        _handle_certificates(nine_nodes_config_dict, None)
+
+    _assert_created_certs(tmp_certs_dir, cluster_manager_certs_dir)
+
+
+def test_credentials_randomly_generated(three_nodes_config_dict):
+    """Test if the credentials are randomly generated and populated.
+
+    In order to test this, we first set the value of arbitrary keys to a
+    fixed value. Then, we test if these keys kept their value and if all other
+    values were populated.
+    """
+    # In this case, The three nodes and nine nodes logic is the same
+    fixed_value = 'fixed_value'
+    chosen_keys = [
+        ('rabbitmq', 'username'),
+        ('postgresql', 'cluster', 'etcd', 'cluster_token'),
+        ('postgresql', 'db_monitoring', 'username'),
+        ('prometheus', 'password')
+    ]
+    credentials_dict = three_nodes_config_dict.get('credentials')
+    _iterate_nested_dict(credentials_dict, chosen_keys, fixed_value)
+    _populate_credentials(credentials_dict)
+    _assert_dict_values_not_none(credentials_dict)
+    _iterate_nested_dict(credentials_dict, chosen_keys, fixed_value, True)
+
+
+def test_config_files_credentials(three_nodes_config_dict, config_files_dir):
+    """Test the credentials in the config files are populated correctly."""
+    # In this case, The three nodes and nine nodes logic is the same
+    credentials_dict = three_nodes_config_dict.get('credentials')
+    _populate_credentials(credentials_dict)
+    _create_config_files(three_nodes_config_dict, config_files_dir,
+                         credentials_dict)
+    _assert_manager_config_credentials(config_files_dir, credentials_dict)
+    _assert_postgresql_config_credentials(config_files_dir, credentials_dict)
+    _assert_rabbitmq_config_credentials(config_files_dir, credentials_dict)
+
+
+def test_ldap_in_config_file(three_nodes_config_dict,
+                             config_files_dir,
+                             ldap_ca_path,
+                             tmp_certs_dir,
+                             cluster_manager_certs_dir):
+    """Test if LDAP is configured properly in the manager config.yaml file."""
+    # In this case, The three nodes and nine nodes logic is the same
+    cluster_manager_ldap_ca = str(cluster_manager_certs_dir / 'ldap_ca.pem')
+    ldap_dict = {
+        'server': 'ldaps://1.1.1.1',
```

Comment:
192.0.2.4 or something would be better.

Reply:
Right, I changed it in the config files but didn't change it here. Changed now.